Server/Components/ObjectOscillator.h

```diff
@@ -0,0 +1,36 @@
+#include "Component.h"
+#include "GameObject.h"
+
+using namespace NCL;
+using namespace CSC8503;
+
+class ObjectOscillator : public Component {
+    enum State {
+        RUNNING,WAITING,COOLDOWN
+    };
+
+public:
+    ObjectOscillator(GameObject* go,float period,float distance,Vector3 direction, float cooldown = 0.0f, float waitDelay = 0.0f);
+    void Update(float dt)override;
+
+private:
+
+    void UpdateOscillation();
+    Vector3 initPosition;
+
+    Vector3 direction; //should be normalised!
```

Comment:
can change variable name to normalisedDirection so that the name is self explanatory